app/models/ems_refresh/save_inventory_container.rb

```diff
@@ -16,20 +30,38 @@ def save_ems_container_inventory(ems, hashes, target = nil)
     ems.save!
   end
 
-  def save_container_projects_inventory(ems, hashes, target = nil)
+  def initialize_inventory_collections(ems)
+    @inv_collections = {}
+    @inv_collections[:container_projects] = ::ManagerRefresh::InventoryCollection.new(
+      :model_class => ContainerProject,
+      :parent => ems,
+      :association => :container_projects
+    )
+    @inv_collections[:container_quotas] = ::ManagerRefresh::InventoryCollection.new(
+      :model_class => ContainerQuota,
+      :parent => ems,
+      :arel => ContainerQuota.joins(:container_project).where('container_projects.ems_id' => ems.id),
+      #:manager_ref => [:ems_ref]
+    )
+    @inv_collections[:container_quota_items] = ::ManagerRefresh::InventoryCollection.new(
+      :model_class => ContainerQuotaItem,
+      :parent => ems,
+      :arel => ContainerQuotaItem.joins(:container_quota => :container_project).where('container_projects.ems_id' => ems.id),
+      # TODO: this builds uuids like "97__cpu", i.e. uses quota id, ideally would use quota ems_ref.
+      #   Can't(?) pass an extra attribute like quota_ems_ref because can't exclude it from saving, manager_ref attributes can't be in attributes_blacklist.
```

Comment:
Right, it's not a problem. It's just that we were talking about building an real/artificial column in each table, that we could use for this and could put there a DB unique constraint.

But it will work as you have it. It's just that there is no lazy_find_by, so it looks a bit clunky. I''ll try to add it later.